packages/wallets/strike/src/adapter.ts

```diff
@@ -0,0 +1,346 @@
+
+    private _connecting: boolean;
+    private _wallet: Window | null;
+    private _isConnected: boolean;
+    private _publicKey: PublicKey | null;
+    private _readyState: WalletReadyState =
+        typeof window === 'undefined' ? WalletReadyState.Unsupported : WalletReadyState.Loadable;
+    private _pendingTransactions: PendingTransactions
+    private _timers: number[]
+
+    constructor() {
+        super();
+        this._connecting = false;
+        this._wallet = null;
+        this._isConnected = false;
+        this._publicKey = null;
+        this._pendingTransactions = {}
+        this._timers = []
+        if (this._readyState !== WalletReadyState.Unsupported) {
+            scopePollingDetectionStrategy(() => {
+                if (this._wallet) {
+                    this._readyState = WalletReadyState.Installed;
```

Comment:
The `scopePollingDetectionStrategy` and `Installed` state are both for adapters that are attached to the browser window. This should be a `Loadable` adapter, and much of its implementation moved to a library.

Reply:
got it